lib/tasks/evm_application.rb

```diff
@@ -114,6 +114,149 @@ def self.output_workers_status(servers)
+        puts "", "For all rows: #{heads_up.map { |n, v| "#{n}=#{v}" }.join(", ")}" if heads_up.present?
+      end
+    end
+
+    def compact_date(date)
+      return "" unless date
+      (date < 1.day.ago) ? date.strftime("%Y-%m-%d") : date.strftime("%H:%M:%S%Z")
+    end
+
+    STATUS_LOOKUP = {
+      "started"        => "Up",
+      "starting"       => "Up",
+      "aborted"        => "Down",
+      "stopped"        => "Down",
+      "not responding" => "?"
+    }
+
+    def compact_status(server)
+      STATUS_LOOKUP[server.status] || server.status || ""
+    end
+
+    # @param sw [MiqServer|MiqWorker] A server or worker object
+    def compact_memory(sw)
+      mb_usage = sw.try(:unique_set_size) || sw.try(:proportional_set_size) || sw.memory_usage
+      return "" unless mb_usage
+      mb_threshold = sw.try(:worker_settings).try("[]", :memory_threshold)
+      mb_threshold ? "#{mb_usage / 1.megabyte}/#{mb_threshold / 1.megabyte}" : (mb_usage / 1.megabyte).to_s
+    end
+
+    def compact_worker_type(name)
+      name && name.gsub(/(ManageIQ::Providers::|Manager|Worker|Miq)/, '')
+    end
+
+    def compact_queue(queue_name, uri)
+      if queue_name.nil?
+        # would like roles to come through
+        if uri
+          uri_parts = uri.split(":")
+          [uri_parts.first, uri_parts.last].join(":")
+        end
+      elsif queue_name.kind_of?(Array)
+        queue_name.join(":")
+      else
+        queue_name
+      end
+    end
+
+    def server_info(s, ws)
+      {
+        "Rgn"        => s.region_number,
+        "Zone"       => s.zone.name,
+        "Status"     => compact_status(s),
+        "Server"     => s.name || "", # s.hostname
+        "Mstr"       => s.is_master? ? "*" : "",
+        "Workers"    => ws ? ws.size : "",
+        "Version"    => s.version,
+#         "Build"      => s.build,
+        "Started On" => compact_date(s.started_on),
+        "Heartbeat"  => compact_date(s.last_heartbeat),
+        "Ram(mb)"    => compact_memory(s),
+        "Roles"      => (s.active_role_names.size == @role_count) ? "*" : s.active_role_names.join(':'),
+      }
+    end
+
+    def servers_overview(servers)
+      servers.collect { |s| server_info(s, s.miq_workers) }
+             .sort_by { |s| [s["Rgn"], s["Zone"], s["Server"]] }
+    end
+
+    def worker_info(s, w)
+      {
+        "Rgn"        => (w || s).region_number,
+        "Zone"       => s && s.zone.name || "?",
+        "Status"     => STATUS_LOOKUP[w.status] || w.status || "",
+        "Server"     => s.try(:name) || w.miq_server_id || "",
+        "Worker"     => compact_worker_type(w.try(:type)) || "MiqServer",
+        "PID"        => (w || s).pid,
+        "SPID"       => (w || s).sql_spid,
+        "Queue"      => compact_queue(w.queue_name, w.uri),
+        "Started On" => compact_date(w.started_on),
+        "Heartbeat"  => compact_date(w.last_heartbeat),
+        "Ram(mb)"    => compact_memory(w),
```

Comment:
if we keep both this and status, we should normalize the language.  We have MB Usage in the `status`.  I'm not opposed to changing it but it should probably be the same in both areas.